source/extensions/http/custom_response/redirect_policy/redirect_policy.cc

```diff
@@ -126,18 +159,26 @@ ::Envoy::Http::FilterHeadersStatus RedirectPolicy::encodeHeaders(
         }
       });
 
+  ::Envoy::Http::Utility::Url absolute_url;
+  std::string uri(uri_ ? *uri_
+                       : ::Envoy::Http::Utility::newPath(*redirect_action_, *downstream_headers));
+  if (!absolute_url.initialize(uri, false)) {
+    stats_.custom_response_invalid_uri_.inc();
+    ENVOY_BUG(!static_cast<bool>(uri_),
```

Comment:
Not sure if this claim is true. IIUC `uri_` is validated, but the call to newPath may return an invalid uri.

Reply:
We're checking here that `uri_` is not set. If it is, we should have errored out earlier during config load.

Reply:
The `if (!absolute_url.initialize(uri, false))` checks `uri` which can be either the member `uri_` or the returned value from `newPath()`, no?

Reply:
We can't check the value of `newPath()` without the request header map, which is not available at config load.

Reply:
I think there's a disconnect between the GFE_BUG checks and what the `if` statement checks. IIUC the `custom_response_invalid_uri` counter will increase, but the ENVOY_BUG may or may not trigger. Is this intentional?

Reply:
I see, thanks for clarifying. Can you please add a comment here?